src/App/Pages/Accounts/TwoFactorPageViewModel.cs

```diff
@@ -208,6 +224,51 @@ public void Load()
             ShowContinue = !(SelectedProviderType == null || DuoMethod || Fido2Method);
         }
 
+        private async Task DuoFramelessAuthenticateAsync()
+        {
+            await _deviceActionService.ShowLoadingAsync(AppResources.Validating);
+            var providerData = _authService.TwoFactorProvidersData[SelectedProviderType.Value];
+            var url = providerData["AuthUrl"] as string;
+            WebAuthenticatorResult authResult;
+            try
+            {
+                authResult = await WebAuthenticator.AuthenticateAsync(new WebAuthenticatorOptions
+                {
+                    Url = new Uri(url),
+                    CallbackUrl = new Uri("bitwarden://duo-callback")
+                });
+            }
+            catch (TaskCanceledException)
+            {
+                // user canceled
+                await _deviceActionService.HideLoadingAsync();
+                return;
+            }
+
+            string response = null;
```

Comment:
🎨 I'd improve the name `response` here to mean something more particular. Like for what I can see is something related with `code` given the logic below, right?